gnat2goto/driver/tree_walk.adb

```diff
@@ -1419,6 +1423,62 @@ package body Tree_Walk is
 
    end Do_Loop_Statement;
 
+   ---------------
+   -- Do_Pragma --
+   ---------------
+
+   procedure Do_Pragma (N : Node_Id; Block : Irep) is
+
+      ----------------------
+      -- Do_Pragma_Assert --
+      ----------------------
+
+      procedure Do_Pragma_Assert (N : Node_Id; Block : Irep);
+      procedure Do_Pragma_Assert (N : Node_Id; Block : Irep) is
+         Assert_Irep : constant Irep := New_Irep (I_Code_Assert);
+
+         --  To be set by iterator:
+         Check : Irep := Ireps.Empty;
+
+         procedure Handle_Arg
+           (Arg_Pos : Positive; Arg_Name : String; Expr : Node_Id);
+         procedure Handle_Arg
+           (Arg_Pos : Positive; Arg_Name : String; Expr : Node_Id) is
+         begin
+            if Arg_Name = "check" or (Arg_Name = "" and Arg_Pos = 1) then
```

Comment:
In GNAT we use short-circuit operators, i.e. `and then` / `or else` (the non-short-circuit `and` and `or` are only OK when both arguments are Boolean variables).